lib/cli.js

```diff
@@ -186,20 +186,24 @@ class Cli extends Base {
           }
         }
 
-        let stack = err.stack ? err.stack.split('\n') : [];
-        if (stack[0] && stack[0].includes(err.message)) {
-          stack.shift();
-        }
+        try {
+          let stack = err.stack ? err.stack.split('\n') : [];
+          if (stack[0] && stack[0].includes(err.message)) {
+            stack.shift();
+          }
 
-        if (output.level() < 3) {
-          stack = stack.slice(0, 3);
-        }
+          if (output.level() < 3) {
+            stack = stack.slice(0, 3);
+          }
 
-        err.stack = `${stack.join('\n')}\n\n${output.colors.blue(log)}`;
+          err.stack = `${stack.join('\n')}\n\n${output.colors.blue(log)}`;
 
-        // clone err object so stack trace adjustments won't affect test other reports
-        test.err = err;
-        return test;
+          // clone err object so stack trace adjustments won't affect test other reports
+          test.err = err;
+          return test;
+        } catch (e) {
+          throw Error(e);
```

Comment:
**question:**
@kobenguyent throwing an error are you sure about it ? i mean the issue about stale process is, i think, throw error, so maybe just log the error and return something else maybe ?

Reply:
Ok i see, so the issue isn't that this code throw an error but is that in case of error nothing is throw and the process didn't get the trigger to stop ?
Seems legit indeed :)
Thanks, i'm trying this version right now, but without retry so i didn't confirm that modif is working